applications/MeshingApplication/custom_utilities/mmg_utilities.cpp

```diff
@@ -0,0 +1,2458 @@
+// KRATOS  __  __ _____ ____  _   _ ___ _   _  ____
+//        |  \/  | ____/ ___|| | | |_ _| \ | |/ ___|
+//        | |\/| |  _| \___ \| |_| || ||  \| | |  _
+//        | |  | | |___ ___) |  _  || || |\  | |_| |
+//        |_|  |_|_____|____/|_| |_|___|_| \_|\____| APPLICATION
+//
+//  License:		 BSD License
+//                       license: MeshingApplication/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+/* The includes related with the MMG library */
+#if !defined(MMG_INCLUDES)
+#define MMG_INCLUDES
+#include "mmg/libmmg.h"
+#include "mmg/mmg2d/libmmg2d.h"
+#include "mmg/mmg3d/libmmg3d.h"
+#include "mmg/mmgs/libmmgs.h"
+#endif /* MMG_INCLUDES defined */
+
+// Project includes
+#include "custom_utilities/mmg_utilities.h"
+
+// NOTE: The following contains the license of the MMG library
+/* =============================================================================
+**  Copyright (c) Bx INP/Inria/UBordeaux/UPMC, 2004- .
+**
+**  mmg is free software: you can redistribute it and/or modify it
+**  under the terms of the GNU Lesser General Public License as published
+**  by the Free Software Foundation, either version 3 of the License, or
+**  (at your option) any later version.
+**
+**  mmg is distributed in the hope that it will be useful, but WITHOUT
+**  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+**  FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+**  License for more details.
+**
+**  You should have received a copy of the GNU Lesser General Public
+**  License and of the GNU General Public License along with mmg (in
+**  files COPYING.LESSER and COPYING). If not, see
+**  <http://www.gnu.org/licenses/>. Please read their terms carefully and
+**  use this copy of the mmg distribution only if you accept them.
+** =============================================================================
+*/
+
+namespace Kratos
+{
+// The member variables related with the MMG library
+MMG5_pMesh mMmgMesh; /// The mesh data from MMG
+MMG5_pSol  mMmgSol;  /// The metric variable for MMG
+// MMG5_pSol  mmgDisp; /// The displacement variable for MMG
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<MMGLibray TMMGLibray>
+void MmgUtilities<TMMGLibray>::PrintMmgMeshInfo(
+    SizeType& rNumberOfNodes,
+    array_1d<SizeType, 2>& rNumberOfConditions,
```

Comment:
Type system could be used here to improve self-documentation of code. Instead of using indices of `array_1d<SizeType, 2>` type you could use a struct of 2x SizeType with meaningful names.

Reply:
That makes a lot of sense, can we do it in a future PR?

Reply:
It's a suggestion, I'm not blocking